frontend/src/app/modules/user-profile/user-profile-page/user-profile.component.ts

```diff
@@ -0,0 +1,219 @@
+
+    public currentUser: UserDto;
+
+    public userNamesForm: FormGroup = new FormGroup({});
+
+    public passwordForm: FormGroup = new FormGroup({});
+
+    public notificationsForm: FormGroup = new FormGroup({});
+
+    constructor(
+        private fb: FormBuilder,
+        private userService: UserService,
+        private notificationService: NotificationService,
+        private authService: AuthService,
+        private spinner: SpinnerService,
+    ) {
+        super();
+    }
+
+    public ngOnInit() {
+        this.fetchCurrentUser();
+    }
+
+    private fetchCurrentUser() {
+        this.spinner.show();
+
+        this.authService
+            .getUser()
+            .pipe(
+                takeUntil(this.unsubscribe$),
+                finalize(() => this.spinner.hide()),
+            )
+            .subscribe(
+                (user) => {
+                    this.currentUser = user as UserDto;
+                    this.initializeForms();
+                },
+                () => {
+                    this.notificationService.error('Failed to fetch current user');
+                },
+            );
+    }
+
+    public override ngOnDestroy() {
+        super.ngOnDestroy();
+    }
+
+    private initializeForms() {
+        this.initUserNamesForm();
+        this.initChangePasswordForm();
+        this.initNotificationsValue();
+    }
+
+    private initUserNamesForm() {
+        this.userNamesForm = this.fb.group({
+            userName: [
+                this.currentUser.userName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.userNameMatch()],
+            ],
+            firstName: [
+                this.currentUser.firstName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.nameMatch()],
+            ],
+            lastName: [
+                this.currentUser.lastName,
+                [Validators.required, Validators.minLength(2), Validators.maxLength(25), ValidationsFn.nameMatch()],
+            ],
+        });
+    }
+
+    private initChangePasswordForm() {
+        this.passwordForm = this.fb.group({
+            currentPassword: ['', [Validators.required, Validators.minLength(2), Validators.maxLength(25)]],
+            newPassword: [
+                '',
+                [
+                    Validators.required,
+                    Validators.minLength(2),
+                    Validators.maxLength(25),
+                    ValidationsFn.lowerExist(),
+                    ValidationsFn.upperExist(),
+                ],
+            ],
+            repeatPassword: [
+                '',
+                [
+                    Validators.required,
+                    Validators.minLength(2),
+                    Validators.maxLength(25),
+                    ValidationsFn.lowerExist(),
+                    ValidationsFn.upperExist(),
+                ],
+            ],
```

Comment:
when I run the app, these validation errors are not shown under the iinputbox, only toastr notifications are working

Reply:
you should move out inputbox to see validation errors

Reply:
i don't see them anyway 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/86236082/8c8bca73-8917-4d37-8716-f0f0bc0decf1)

Reply:
also, use `<app-input [type]="password">`, don't create your own implementation, it is already done for you